src/contexts/ServerContext.js

```diff
@@ -428,10 +428,27 @@ export const ServerProvider = ({ children }) => {
     const loadServers = async () => {
         setIsLoadingZoneContext(true);
         let zone_report = await loadZoneReport();
+
         if (zone_report !== undefined) {
             let resc_types = new Set();
-            let catalog_service_provider = [zone_report.data.zones[0]['icat_server']];
-            let fullServersArray = catalog_service_provider.concat(zone_report.data.zones[0]['servers']);
+            let version = zone_report.data.zones[0].servers[0]?.version?.irods_version;
+            let diffFrom4_3_1
+            try {
+                // 4.3.1 is a new release of schema after 4.1.0; this code handles both versions
+                diffFrom4_3_1 = irodsVersionComparator(version, "4.3.1"); // if version > 4.3.1, returns positive integer, if equal to 4.3.1, returns 0, if less than 4.3.1, returns negative
+            } catch(e) {
+                diffFrom4_3_1 = -1; // on versions before 4.3.1, version will be undefined
+            }
+            
+            let fullServersArray;
+
+            if (diffFrom4_3_1 >= 0) {
+                // after 4.3.1, icat_server is not separate anymore, it is now included in servers array
+                fullServersArray = zone_report.data.zones[0]['servers'];
+            } else {
+                let catalog_service_provider = [zone_report.data.zones[0]['icat_server']];
+                fullServersArray = catalog_service_provider.concat(zone_report.data.zones[0]['servers']);
```

Comment:
should this be ... `extend` rather than `concat`?

Reply:
Pretty sure it should be concat ([mdn docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat)). I'm pretty sure `extend` is used for Python lists

Reply:
ha, javascript.   silly me.  carry on.